scripts/enforce-change-notes-updated-on-pr-to-master

```diff
@@ -20,13 +15,11 @@ if [[ ${CI_PULL_REQUEST-} ]]; then
 
   # PRs to master are typically either release or hotfix PRs, both of which should have change notes updated
   if [[ $base_branch == master ]]; then
-    head_notes=$(change_notes_from_git_revision HEAD)
-    # CircleCI fetches all the branches from the repo in the Checkout step (but only into refs/remotes, not into refs/heads);
-    # thus, origin/master should be available.
-    master_notes=$(change_notes_from_git_revision origin/master)
-
-    if [[ "$head_notes" == "$master_notes" ]]; then
-      die "Content of $change_notes_path has not been changed from the current master (commit $(git rev-parse --short origin/master))"
+    if ! ./gradlew getChangelog --no-header -q &> /dev/null; then
+      echo "Content of $change_notes_path has not been patched, fill the 'Unreleased' section and run ./gradlew patchChangelog"
```

Comment:
Note that before the actual release (which pushes the tag to GitHub, see .circleci/config.yml:218), `patchChangelog` will move the contents of `Unreleased` section to sth like `## v3.0.0-37-SNAPSHOT+git.e41777f-dirty`... while we'd expect `## v3.0.0`. Pls reflect that in the error message

Reply:
Actually, once I've run `patchChangelog` from the state on this PR, and then run `./gradlew getChangelog --no-header -q`, I've ended up with:

```

FAILURE: Build failed with an exception.

* Where:
Build file '/home/plipski/git-machete-intellij-plugin/build.gradle.kts' line: 199

* What went wrong:
Header 'v3.0.0-37-SNAPSHOT+git.e41777f-dirty' does not contain version number. 
```

since a snapshot version format doesn't match `headerParserRegex` :/